File: algorithms/leetcode/1064-fixed-point/FixedPoint.java
class Solution {
    public int fixedPoint(int[] A) {
        int l = 0;
        int r = A.length - 1;
        
        int result = -1;
        
        while (l <= r) {
            int mid = l + (r - l) / 2;
            if (A[mid] >= mid) {
                if (A[mid] == mid) {
                    result = mid;
                }
                
                r = mid - 1;
            } else {
                l = mid + 1;
            }
        }
        
        return result;
    }
}

File: leetcode/1064-fixed-point/FixedPoint.cs
public class Solution {
    public int FixedPoint(int[] A) {
        int left = 0;
        int right = A.Length - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            if (A[mid] == mid) {
                return mid;
            } else if (A[mid] > mid) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }

        return -1;
    }
}
